swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
 permalink: /learn/edi-tool/
 active: edi-tool
 intro: The EDI tool provides the below set of command line tools to work with EDI files in Ballerina.
----
+--- 
+
+- **Code generation:** Generate Ballerina records and parsing functions for a given EDI schema
+- **Library generation:** Generate Ballerina records, parsing functions, utility methods, and a REST connector for a given collection of EDI schemas, and also, organize those as a Ballerina library
+
+## Compatibility
+
+|                            | Compatible with |
+|----------------------------|------------------|
+| Ballerina language version | 2201.7.0         |
+| Java Development Kit (JDK) | 11               |
+
+## Usage
+
+The usage of the tool is described below.
+
+```
+bal edi codegen <EDI schema path> <output path>
+```
+
+The above command generates all Ballerina records and parsing functions required for working with data in the given EDI schema and writes those into the file specified in the output path. The generated parsing function (i.e., `fromEdiString(...)`) can read EDI text files into generated records, which can be accessed from the Ballerina code similar to accessing any other Ballerina record. Similarly, the generated serialization function (i.e., `toEdiString(...)`) can serialize generated Ballerina records into EDI text.
+
+## Example
+
+A simple EDI schema (`edi-schema.json` file) is shown below.
+
+```json
+{
+    "name": "SimpleOrder",
+    "delimiters" : {"segment" : "~", "field" : "*"},
+    "segments" : {
+        "HDR": {
+            "tag" : "header",
+            "fields" : [{"tag" : "code"}, {"tag" : "orderId"}, {"tag" : "organization"}, {"tag" : "date"}]
+        },
+        "ITM": {
+            "tag" : "items",
+            "maxOccurances" : -1,
+            "fields" : [{"tag" : "code"}, {"tag" : "item"}, {"tag" : "quantity", "dataType" : "int"}]
+        }
+    }
+}
+```
+
+The above schema can be used to parse EDI documents with one HDR segment (mapped to `header`) and any number of ITM segments (mapped to `items`). The HDR segment contains three fields, which are mapped to `orderId`, `organization`, and `date`. Each ITM segment contains two fields mapped to `item` and `quantity`. Below is a sample EDI document that can be parsed using the above schema (`edi-sample1.edi` file).
```

Comment:
Do we need these file names here? `edi-sample1.edi` and also it is a bad file name with number 1.